s3/replication/common/src/s3replicationcommon/s3_upload_part.py

```diff
@@ -0,0 +1,157 @@
+        query_params = urllib.parse.urlencode(
+            {'partNumber': self._part_no, 'uploadId': self._upload_id})
+        body = ""
+
+        headers = AWSV4Signer(
+            self._session.endpoint,
+            self._session.service_name,
+            self._session.region,
+            self._session.access_key,
+            self._session.secret_key).prepare_signed_header(
+            'PUT',
+            request_uri,
+            query_params,
+            body)
+
+        if (headers['Authorization'] is None):
+            self._logger.error(fmt_reqid_log(self._request_id) +
+                               "Failed to generate v4 signature")
+            sys.exit(-1)
+
+        headers["Content-Length"] = str(chunk_size)
+
+        self._logger.info(fmt_reqid_log(self._request_id) +
+                          "PUT on {}".format(
+                              self._session.endpoint + request_uri))
+        self._logger.debug(fmt_reqid_log(self._request_id) +
+                           "PUT with headers {}".format(headers))
+
+        self._timer.start()
+        try:
+            async with self._session.get_client_session().put(
+                    self._session.endpoint + request_uri,
+                    headers=headers,
+                    params=query_params,
+                    data=data_reader.fetch(chunk_size)) as resp:
+                self._timer.stop()
+
+                self._http_status = resp.status
+                self._response_headers = resp.headers
+
+                self._logger.info(
+                    fmt_reqid_log(self._request_id) +
+                    'PUT Object completed with http status: {}'
+                    '\n header{}'.format(
+                        resp.status, self._response_headers))
+
+                self._etag_dict[self._part_no] = self._response_headers["Etag"]
+
+                if resp.status == 200:
+                    self._state = S3RequestState.COMPLETED
+                else:
+                    error_msg = await resp.text()
+                    self._logger.error(
+                        fmt_reqid_log(self._request_id) +
+                        'Error Response: {}'.format(error_msg))
+                    self._state = S3RequestState.FAILED
+        except aiohttp.client_exceptions.ClientConnectorError as e:
+            self._timer.stop()
+            self.remote_down = True
+            self._state = S3RequestState.FAILED
+            self._logger.error(fmt_reqid_log(self._request_id) +
+                               "Failed to connect to S3: " + str(e))
+        return
+
+    def pause(self):
+        self._state = S3RequestState.PAUSED
+        # XXX Take real pause action
+
+    def resume(self):
+        self._state = S3RequestState.PAUSED
+        # XXX Take real resume action
+
+    def abort(self):
+        self._state = S3RequestState.ABORTED
+        # Abort the reader so that PUT can stop.
+        self._data_reader.abort()
```

Comment:
Can be removed as not being used or contributing to actual funcationality.

Reply:
sure. done